src/routes/api/maintenance/copyPreloadedDecks/+server.ts

```diff
@@ -0,0 +1,65 @@
+//@ts-nocheck
+import type { RequestHandler } from './$types';
+import { db, readPath, writePath } from '$lib/server/firebaseUtils';
+
+const createDeckCopy = async (deckId: number, userId: string) => {
+  // Generate new reference ID
+  const newRefId = Math.floor(Math.random() * 4294967295);
+
+  // Fetch linked deck
+  const linkedDeck = await readPath<Database.Decks.Preloaded>(`/decks/preloaded/${deckId}`);
+  if (linkedDeck) {
+    // Create a copy of the linked deck with updated fields
+    const timestamp = new Date().toISOString().split('.')[0];
+    const deckCopy = { ...linkedDeck };
+    deckCopy.created_ts = timestamp;
+    deckCopy.modified_ts = timestamp;
+    deckCopy.position = -1;
+    deckCopy.refId = newRefId;
+    deckCopy.is_editable = true;
+
+    // Save the copy to the user's decks
+    await writePath<Database.Decks.User>(`/decks/user/${userId}/${newRefId}`, deckCopy);
+  }
+};
+
+export const POST: RequestHandler = async (event) => {
+  const playlists = (await db.ref('/playlists/user').get()).val();
+
+  const counts: { [userId: string]: { [id: string]: number } } = {};
+
+  if (playlists) {
+    Object.keys(playlists).forEach(async (userId) => {
+      counts[userId] = { '113822357': 0, '541852177': 0 }; // Initialize counts for each user
+      const userPlaylists = playlists[userId];
+
+      Object.keys(userPlaylists).forEach(async (playlistId) => {
+        const playlist = userPlaylists[playlistId] as PlaylistData;
+        if (playlist.linked_deck_id === 113822357 || playlist.linked_deck_id === 541852177) {
+          counts[userId][playlist.linked_deck_id]++;
+        }
+      });
+    });
+  }
```

Comment:
Both `forEach` loops with an async function will not block, so execution will keep going. This is the kind of thing that may work on local but will fail in a serverless environment because one the top-level `POST` function returns there are no handles for the serverless function to wait on so it will just shut off. I am not seeing anything asynchronous here so I think we can just make the callback in the for loop regular functions